File: date/csharp/src/ApiDateHelper.cs
using System;

namespace bayeslife.apiStandardization
{
/*  */
public class  ApiDateHelper  {

      public DateTime propertyToDate(string date){
            DateTime dt = DateTime.Parse(date);
            return dt;
            
      }
      
       public String dateToProperty(DateTime date){
             return date.ToString("yyyy-MM-dd");
            
      }

}
}

File: date/java/src/main/java/bayeslife/ApiDateHelper.java
package bayeslife;

import java.text.SimpleDateFormat;
import java.time.LocalDate;
import java.util.Date;

/**
 *
 */
public class ApiDateHelper
{

    SimpleDateFormat format = new SimpleDateFormat();

    public LocalDate propertyAsDate(String date_property){
        //This returns date in local time zone
        LocalDate date = LocalDate.parse(date_property);
        return date;
    }

    public String dateAsProperty(LocalDate date){
        //Converts javascript date in UTC into date
        return date.toString();
        //return format.format(date);
    }

    public String dateAsProperty(Date date){
        //Converts javascript date in UTC into date
        return format.format(date);
    }

    public String dateAsISO(Date date) {
        return format.format(date);
    }
}
